woocommerce/woocommerce-ios
# Dynamic Dashboard: Store performance card header

Part of #12465 


## Description

This PR starts the implementation for the store performance card in SwiftUI. Due to the large change, only the card header and the sync logic is handled - the rest will be taken care of in a separate PR.

Changes include:
- Disabled the feature flag `dynamicDashboard` for UI tests to avoid failures in the stats screen. We'll need to update the tests in a separate task.
- Replaced the legacy view controller with the new store performance card view.
- Added a reusable view `StatsTimeRangePicker` to use for the top performer card as well.
- Added `StorePerformerView` with just a title and time range picker view.
- Added `StorePerformerViewModel` with the logic for syncing data, time range picker, and time range text. 

Unit tests will be added in a separate PR due to the large number of changes.

## Testing instructions

- Enable the feature flag `dynamicDashboard` and build the app.
- Confirm that the WIP store performer card is displayed if the card is enabled.
- Tap the calendar icon and select a different time range - confirm that the card is reloaded and then updated with the correct time range text.
- Disable the Internet connection and pull to refresh the dashboard screen. Confirm that the error banner is displayed after syncing fails.
- Enable the connection again and pull to refresh the dashboard screen. Confirm that the error banner is dismissed and the correct time range text is displayed again.

## Screenshots


Normal state | Error state | Loading state
--- | --- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/f99201f2-9c33-4f4e-871c-5e07dceaf95b" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/70809a5e-0769-4fce-a23e-cf738dc54227" width=320 /> |  <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/56116164-d6d2-4376-b082-34aa2e4a3910" width=320 /> 

---
